modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro

```diff
@@ -7,16 +7,96 @@ definition {
 
 		Click(
 			key_column = "${key_column}",
-			locator1 = "FrontendDataSet#SELECT_FIELD");
+			locator1 = "FrontendDataSet#OPEN_SELECT_FIELD");
 
 		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
 	}
 
+	macro changeCustomView {
+		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
+
+		Click(
+			key_optionName = "${key_optionName}",
+			locator1 = "FrontendDataSet#SELECT_OPTION");
+	}
+
+	macro compareValue {
+		var dataSetValue1 = FDSCustomView.getValueByPosition(
+			columnIndex = "${key_columnIndex}",
+			valueIndex = "${key_valueIndex1}");
+		var dataSetValue2 = FDSCustomView.getValueByPosition(
+			columnIndex = "${key_columnIndex}",
+			valueIndex = "${key_valueIndex2}");
+
+		TestUtils.isGreaterThan(
+			actual = "${dataSetValue1}",
+			expected = "${dataSetValue2}");
+	}
+
 	macro createNewView {
 		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
 
 		Click(
 			key_itemName = "Save View As",
+			key_optionName = "Save View As",
+			locator1 = "FrontendDataSet#SELECT_OPTION");
+
+		Type(
+			locator1 = "FrontendDataSet#INPUT_CUSTOM_VIEW_NAME",
+			value1 = "${key_nameView}");
+
+		Click(
+			key_name = "Save",
+			locator1 = "Button#GENERAL_BUTTON_MODAL_FOOTER");
+	}
+
+	macro editFilters {
+		FDSFilters.openFilters();
+
+		FDSFilters.searchFilter(searchTerm = "${key_filter}");
+
+		Click(
+			key_filter = "${key_filter}",
+			locator1 = "FrontendDataSet#FILTER_SUMMARY_LABEL");
+
+		FDSFilters.disableStatusFilters(key_status = "${key_disableFilters}");
+
+		FDSFilters.enableStatusFilters(key_status = "${key_enableFilters}");
+
+		Click(locator1 = "FrontendDataSet#EDIT_FILTER_BUTTON");
+
+		Click(locator1 = "FrontendDataSet#FILTER_BUTTON");
+
+		FDSFilters.closeFilters();
+	}
+
+	macro getValueByPosition {
+		var key_columnIndex = "${columnIndex}";
+		var key_valueIndex = "${valueIndex}";
+		var dataSetValue = selenium.getText("FrontendDataSet#TABLE_CELL");
+
+		return "${dataSetValue}";
+	}
+
+	macro saveCustomView {
+		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
+
+		Click(
+			key_optionName = "Save View",
+			locator1 = "FrontendDataSet#SELECT_OPTION");
+	}
+
+	macro saveView {
+		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
+
+		Click(locator1 = "FrontendDataSet#SAVE_BUTTON");
+	}
+
```

Comment:
Could you put these two macros (saveView and saveCustomView) together in one using a parameter to do both functions?